### PR TITLE
add dob check to crew eligibility

### LINKED ISSUE #55: DOB validation for sign ups
Require person be 18 to sign up for night crew. Allow for sign up for games/events but with a different name style on event page.

### PR DESCRIPTION
**Pull request by ddbruce**
Should fix #55

### CODE REVIEW

File: .crews.php [hunk 1]
```diff
@@ -47,9 +47,16 @@ function clearCrew ($connection, $signature, $id, $position, $crewid) {
     modifyCrewAssignment($connection, $signature, $id, $position, $crewid, 'clear');
 }
 
-function determineEligibility ($member, $pos, $i, $ontoday, $onthisweek, $ccton, $dton, $atton, $obson, $y, $m, $d) {
+function determineEligibility ($member, $pos, $i, $ontoday, $onthisweek, $ccton, $dton, $atton, $obson, $y, $m, $d, $dob) {
+    $dob = explode("-", $dob);
+    $age = (date("md", date("U", mktime(0, 0, 0, $dob[0], $dob[1], $dob[2]))) > date("md")
+      ? ((date("Y") - $birthDate[2]) - 1)
+      : (date("Y") - $birthDate[2]));
```
Comment (MasterOdin) [inline .crews.php]: `$birthDate` should be `$dob`?

File: .crews.php [hunk 2]
```diff
@@ -47,9 +47,16 @@ function clearCrew ($connection, $signature, $id, $position, $crewid) {
     modifyCrewAssignment($connection, $signature, $id, $position, $crewid, 'clear');
 }
 
-function determineEligibility ($member, $pos, $i, $ontoday, $onthisweek, $ccton, $dton, $atton, $obson, $y, $m, $d) {
+function determineEligibility ($member, $pos, $i, $ontoday, $onthisweek, $ccton, $dton, $atton, $obson, $y, $m, $d, $dob) {
+    $dob = explode("-", $dob);
```
Comment (MasterOdin) [inline .crews.php]: You probably want to change this to be `$dob = explode('-', $member['dob']);` and remove `$dob` from the function signature as `$dob` is undefined (and you'd need to pass `$dob` to `populateSpot(..)` and then from there pass `$dob` into `determineEligibility()`, but it's just going to be `$member['dob']`.

File: .crews.php [hunk 3]
```diff
@@ -47,9 +47,16 @@ function clearCrew ($connection, $signature, $id, $position, $crewid) {
...
+    $age = (date("md", date("U", mktime(0, 0, 0, $dob[0], $dob[1], $dob[2]))) > date("md")
+      ? ((date("Y") - $dob[2]) - 1)
+      : (date("Y") - $dob[2]));
```
Comment (MasterOdin) [inline .crews.php]: The format of the `dob` field is `YYYY-MM-DD` so your array keys need to be reversed (replace `0` with `2` and vice versa).

**Comment by ddbruce** [comment]
Feel free to make changes directly to the branch, @MasterOdin. I agree with your assessment about this only disabling a button; we have a lot of poor security like this in the site. I don't mean to further it, but this is a quick fix to get the functionality I think we're looking for.

**Pull request merged**